src/main/java/org/cryptomator/cryptofs/fh/FileHeaderHolder.java

```diff
@@ -29,33 +31,41 @@ public FileHeaderHolder(Cryptor cryptor, @CurrentOpenFilePath AtomicReference<Pa
 	}
 
 	public FileHeader get() {
-		FileHeader result = header.get();
-		if (result == null) {
-			throw new IllegalStateException("Header not set.");
-		}
-		return result;
+		return header.get();
```

Comment:
can we be sure, this is never called prematurely? I would rather have an explicit exception than a NPE